src/features/generic_events/aggregate/user-actions/utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright 2020-2025 New Relic, Inc. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Checks if the element is visible in the DOM
+ * @param elem
+ * @returns {boolean|boolean}
+ */
+export function isVisible (elem) {
```

Comment:
is it possible to "click" on an element that is hidden with these criteria?

Reply:
Would it be simpler to check the event's `.isTrusted` property before running the DOM tree parsing?